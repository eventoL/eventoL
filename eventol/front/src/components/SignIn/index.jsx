import React from 'react';

import DOM from '../../utils/dom';
import {LOGIN_URL, LOGOUT_URL} from '../../utils/urls';

import './index.scss';


export default class SignIn extends React.PureComponent {
  searchFocus = () => {
    DOM.focusOn('search-input');
  }

  render(){
    return (
      <div className='sign-in'>
        <ul>
          <li><a href={LOGIN_URL}>{gettext('Sign In')}</a></li>
          <li><a href={LOGOUT_URL}>{gettext('Sign Up')}</a></li>
        </ul>
      </div>
    );
  }
}
