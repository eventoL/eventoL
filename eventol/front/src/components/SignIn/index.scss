@import '../../styles/variables';
@import '../../styles/commons';

@media (min-width: 950px) {
  .sign-in {
    display: block;
    position: absolute;
    right: 20px;
  
    ul {
      @include flex-element;
      @include center-element;
      height: 37px;
      padding: 0 10px;
  
      li {
        -webkit-transition: background .125s ease;
        border-radius: 3px;
        font-size: 20px;
        font-weight: 400;
        margin-right: 20px;
        padding: 7px 10px;
        transition: background .125s ease;
  
        a {
          text-decoration: none;
        }
      }
  
      li:hover {
        background: $eventol-color;
        cursor: pointer;
      }
    }
  }
}

@media (max-width: 950px) {
  .sign-in {
    ul {
      background: #666a73;
      margin-right: 8vw;
      opacity: .8;
      z-index: 0;

      a {
        float: unset !important;
        text-decoration: none;

        li:hover {
          background: $eventol-color;
          cursor: pointer;
        }

        li {
          -webkit-transition: background .125s ease;
          border-radius: 3px;
          font-size: 14px;
          font-weight: 400;
          padding: 7px 10px;
          text-align: left;
          transition: background .125s ease;
        }
      }

      li {
        -webkit-transition: background .125s ease;
        border-radius: 3px;
        font-size: 14px;
        font-weight: 400;
        padding: 7px 10px;
        text-align: left;
        transition: background .125s ease;
      }

      li:hover {
        background: $eventol-color;
        cursor: pointer;
      }
    }
  }
}
