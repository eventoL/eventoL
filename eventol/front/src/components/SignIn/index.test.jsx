import React from 'react';
import renderer from 'react-test-renderer';

import SignIn from '.';

describe('SignIn', () => {
  let component;
  let tree;

  const getComponent = () => {
    component = renderer.create(<SignIn />);
    return component;
  };

  beforeEach(() => {
    component = getComponent();
    tree = component.toJSON();
  });

  describe('Default', () => {
    test('Snapshot', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
