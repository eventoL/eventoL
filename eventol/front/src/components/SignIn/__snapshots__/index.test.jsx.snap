// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Default Snapshot 1`] = `
<div
  className="sign-in"
>
  <ul>
    <li>
      <a
        href="/accounts/login/"
      >
        Sign In
      </a>
    </li>
    <li>
      <a
        href="/accounts/signup/"
      >
        Sign Up
      </a>
    </li>
  </ul>
</div>
`;
